src/app/components/sites/site/site.component.ts

```diff
@@ -22,17 +26,84 @@ class SiteComponent extends PageComponent implements OnInit {
   @Input() public site: Site;
   public defaultDescription = "<i>No description found</i>";
   public recordings: AudioRecording[];
-  public recordingsEnd: DateTimeTimezone;
-  public recordingsStart: DateTimeTimezone;
+  public recordingsEnd: DateTime;
+  public recordingsStart: DateTime;
   public marker: List<MapMarkerOption>;
+  public annotations: AudioEvent[];
 
-  constructor() {
+  constructor(
+    private audioEventsApi: ShallowAudioEventsService,
+    private audioRecordingsApi: AudioRecordingsService
+  ) {
     super();
   }
 
   public ngOnInit() {
-    this.recordings = [];
     this.marker = sanitizeMapMarkers(this.site.getMapMarker());
+
+    this.getAnnotations();
+    this.getNewestDates();
+    this.getOldestDates();
+  }
+
+  public humanizeDate(date: DateTime) {
+    return date ? date.toLocaleString(DateTime.DATETIME_FULL) : "(loading)";
+  }
+
+  private getAnnotations() {
+    this.audioEventsApi
+      .filter({
+        filter: { ["audio_recordings.site_id"]: { eq: this.site.id } } as any,
+        sorting: { orderBy: "updatedAt", direction: "desc" },
+      })
+      .pipe(takeUntil(this.unsubscribe))
+      .subscribe(
+        (events) => {
+          // Limit the selection of audio events by tagging count
+          let count = 0;
+          const maxTags = 10;
+          this.annotations = [];
+          for (const event of events) {
+            this.annotations.push(event);
+            count++;
+
+            if (count > maxTags) {
+              return;
+            }
+          }
+        },
+        (err) => console.log({ err })
+      );
+  }
+
+  private getNewestDates() {
+    this.audioRecordingsApi
+      .filter({
+        filter: { ["sites.id"]: { eq: this.site.id } } as any,
+        sorting: { orderBy: "recordedDate", direction: "desc" },
+      })
+      .pipe(takeUntil(this.unsubscribe))
+      .subscribe(
+        (recordings) => {
+          this.recordingsEnd = recordings[0]?.recordedDate;
+          this.recordings = recordings;
+        },
+        (err) => console.log({ err })
+      );
+  }
+
+  private getOldestDates() {
+    this.audioRecordingsApi
+      .filter({
+        filter: { ["sites.id"]: { eq: this.site.id } } as any,
+        sorting: { orderBy: "recordedDate", direction: "asc" },
+        paging: { items: 1 },
+      })
+      .pipe(takeUntil(this.unsubscribe))
+      .subscribe(
+        (recordings) => (this.recordingsStart = recordings[0]?.recordedDate),
+        (err) => console.log({ err })
+      );
```

Comment:
generally: i'd like to see service queries defined in the service files. E.g. `this.audioRecordingsApi.getNewestDates(siteId)`

We haven't established this yet, but let's consdier that the new standard

Reply:
Ill turn this into a separate PR